pkg/pool-weighted/contracts/smart/IndexPool.sol

```diff
@@ -46,15 +48,20 @@ contract IndexPool is BaseWeightedPool, ReentrancyGuard {
 
     // Store scaling factor and start/end weights for each token
     // Mapping should be more efficient than trying to compress it further
-    // [ 155 bits|   5 bits |  32 bits   |   64 bits    ]
-    // [ unused  | decimals | end weight | start weight ]
+    // [ 123 bits| 32 bits        |   5 bits |  32 bits   |   64 bits    ]
+    // [ unused  | target weights | decimals | end weight | start weight ]
```

Comment:
Do we need target weight?

Reply:
On the token addition we setup initial weight 1%, we can add initialized parameter which tells us if token has reached 1% and can increase its weight up to end weight....

Reply:
I think end weight and target weight is already the same thing

Reply:
We need it for when the new token get's initialized at some point (due to swaps), to know which was the original final weight for the new token.

Reply:
the end weight is the currently applicable end weight for a token, that is also used to calc the current weight of a token. The target weight is not used to calc the price, it will be set as end weight as soon as the new token becomes initialized.

Reply:
Hmm, this makes sense, I just thought maybe we can make it the other way algorithmically. Though I think changing algo would be a suicide rn in terms of time, so makes sense.